compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -37,6 +38,8 @@ import annotation.{constructorOnly, threadUnsafe}
 object Applications {
   import tpd._
 
+  private val FlowFactsOnTree = new Property.Key[FlowTyper.FlowFacts]
```

Comment:
This we should probably be moved to Typer.scala, where other typer-related properties are defined.

Reply:
Also, could you please add a comment explaining what this does?